src/js/components/Booking.js

```diff
@@ -226,10 +225,7 @@ class Booking{
       body: JSON.stringify(payload)
     };
 
-    fetch(url, options);
-    thisBooking.makeBooked(payload.date, payload.hour, payload.duration, payload.table);
-    // thisBooking.updateDOM();
-    // console.log(payload);
+    fetch(url, options).then(thisBooking.makeBooked(payload.date, payload.hour, payload.duration, payload.table));
```

Comment:
Musisz to wywołanie jeszcze opakować w funkcję anonimową. Teraz to wywołujesz tę funkcję od razu, a jako callback przekazujesz to co zwraca metoda makeBooked